gammapy/spectrum/flux_point.py

```diff
@@ -18,12 +19,379 @@
+        except KeyError:
+            x_err = None
+        return x_err
+
+    def _plot_get_y_err(self, sed_type):
+        try:
+            # assymmetric error
+            y_errn = self.table[sed_type + '_errn'].quantity
+            y_errp = self.table[sed_type + '_errp'].quantity
+            y_err = (y_errn, y_errp)
+        except KeyError:
+            try:
+                # symmetric error
+                y_err = self.table[sed_type + '_err'].quantity
+                y_err = (y_err, y_err)
+            except KeyError:
+                # no error at all
+                y_err = None
+        return y_err
+
+    @property
+    def _is_ul(self):
+        try:
+            return self.table['is_ul'].data.astype('bool')
+        except KeyError:
+            return np.isnan(self.table[self.sed_type])
+
+    def show(self, figsize=(8, 5), **kwargs):
+        """
+        Show flux points.
+
+        Parameters
+        ----------
+        figsize : tuple
+            Figure size
+        kwargs : dict
+            Keyword arguments passed to `FluxPoints.plot()`.
+
+        Returns
+        -------
+        ax : `~matplotlib.axes.Axes`
+            Plotting axes object.
+        """
+        import matplotlib.pyplot as plt
+
+        fig = plt.figure(figsize=figsize)
+        ax = fig.add_subplot(111)
+        self.plot(ax=ax, **kwargs)
+        return ax
+
+    def __str__(self):
+        """
+        String representation of the flux points class.
+        """
+        info = ''
+        info += "Flux points of type '{}'".format(self.sed_type)
+        return info
+
+    def info(self):
+        """
+        Print flux points info.
+        """
+        print(self)
+
+    @classmethod
+    def read(cls, filename, **kwargs):
+        """
+        Read flux points.
+
+        Parameters
+        ----------
+        filename : str
+            Filename
+        kwargs : dict
+            Keyword arguments passed to `~astropy.table.Table.read`.
+
+        Examples
+        --------
+
+        >>> from gammapy.spectrum import FluxPoints
+        >>> filename = '$GAMMAPY_EXTRA/test_datasets/spectrum/flux_points/flux_points.fits'
+        >>> flux_points = FluxPoints.read(filename)
+        >>> flux_points.show()
+
+        """
+        filename = make_path(filename)
+        try:
+            table = Table.read(str(filename), **kwargs)
+        except IORegistryError:
+            kwargs.setdefault('format', 'ascii.ecsv')
+            table = Table.read(str(filename), **kwargs)
+
+        if 'SED_TYPE' not in table.meta.keys():
+            sed_type = cls._guess_sed_type(table)
+            table.meta['SED_TYPE'] = sed_type
+
+        return cls(table=table)
+
+    def write(self, filename, **kwargs):
+        """
+        Write flux points.
+
+        Parameters
+        ----------
+        filename : str
+            Filename
+        kwargs : dict
+            Keyword arguments passed to `~astropy.table.Table.write`.
+        """
```

Comment:
I suggest to also pass `filename` through:
```
filename = make_path(filename)
```
It does e.g. os environ variable expansion, which is something I find convenient and use frequently.